client/tests/integration/queries/asset.rs

```diff
@@ -0,0 +1,260 @@
+            10.0_f64.try_to_value()?,
+            5.0_f64.try_to_value()?,
+            NumericValue::Fixed(Fixed::try_from(30.0)?),
+        ),
+    ] {
+        // Registering new asset definition
+        let definition_id: <AssetDefinition as Identifiable>::Id =
+            definition.parse().expect("Failed to parse `definition_id`");
+        let asset_definition = AssetDefinition::new(definition_id.clone(), asset_value_type);
+        test_client.submit_blocking(RegisterBox::new(asset_definition.clone()))?;
+
+        let assets = accounts
+            .iter()
+            .cloned()
+            .map(|account_id| <Asset as Identifiable>::Id::new(definition_id.clone(), account_id))
+            .collect::<Vec<_>>();
+
+        // Give Alice ability to burn other accounts assets
+        assets
+            .iter()
+            .skip(1)
+            .cloned()
+            .map(|asset_id| CanBurnUserAssets::new(asset_id).into())
+            .map(|permission_token: PermissionToken| {
+                GrantBox::new(permission_token, accounts[0].clone())
+            })
+            .zip(accounts.iter().skip(1).cloned())
+            .zip(keys.iter().cloned())
+            .map(|((grant_box, account_id), key_pair)| {
+                Transaction::new(
+                    account_id,
+                    Executable::Instructions(vec![grant_box.into()]),
+                    100_000,
+                )
+                .sign(key_pair)
+                .expect("Signing failed")
+            })
+            .for_each(|transaction| {
+                test_client
+                    .submit_transaction_blocking(transaction)
+                    .expect("Unable to execute transaction");
+            });
+
+        // Assert that initial total quantity before any burns and mints is zero
+        let initial_total_asset_quantity = test_client.request(
+            FindTotalAssetQuantityByAssetDefinitionId::new(definition_id.clone()),
+        )?;
+        assert!(initial_total_asset_quantity.is_zero_value());
+
+        let register_asset = assets
+            .iter()
+            .cloned()
+            .map(|asset_id| Asset::new(asset_id, initial_value.clone()))
+            .map(|asset| RegisterBox::new(asset).into())
+            .collect::<Vec<_>>();
+        test_client.submit_all_blocking(register_asset)?;
+
+        let mint_asset = assets
+            .iter()
+            .cloned()
+            .map(|asset_id| MintBox::new(to_mint.clone(), asset_id).into());
+        test_client.submit_all_blocking(mint_asset)?;
+
+        let burn_asset = assets
+            .iter()
+            .cloned()
+            .map(|asset_id| BurnBox::new(to_burn.clone(), asset_id).into())
+            .collect::<Vec<_>>();
+        test_client.submit_all_blocking(burn_asset)?;
+
+        // Assert that total asset quantity is equal to: `sum(initial_values) + sum(mints) - sum(burns)`
+        let total_asset_quantity = test_client.request(
```

Comment:
This was a bit confusing to me on first read. Took a while to realize that those initial values are repeated 5 times for each respective account.